### DISCUSSION TITLE
Feature Request: Setting Postgres statement timeout

### DISCUSSION THREAD

**Discussion by noppaz** [duckdb/duckdb#10550] [Ideas]
When attaching a Postgres database, I would like to be able to adjust the statement timeout on the Postgres database connection. The timeout can be adjusted for a Postgres user or role by the database admin, but allowing session specific adjustments enables non admins to adjust this per session to prevent things such as timeouts when copying a table from Postgres through Duckdb.

If I used a Python Postgres adapter, I would run the `SET statement_timeout = X` query in the connection session to adjust this. This could be another [setting](https://duckdb.org/docs/extensions/postgres.html#settings) in Duckdb when attaching a Postgres db.

**Comment by noppaz** [comment]
Posted another idea [here](https://github.com/duckdb/duckdb/discussions/10712) with a more general use case, which could solve this specific one.

**Comment by amitsaxena** [comment]
We ran into this issue for a long running query and it's inconvenient that Duckdb doesn't allow you to control database timeout from the client. This can be a really useful feature for several use cases.
